core/network/src/consensus_gossip.rs

```diff
@@ -357,14 +324,52 @@ impl<B: BlockT> ConsensusGossip<B> {
 					entry.remove_entry();
 				}
 			}
-			self.multicast_inner(protocol, message_hash, topic, status, || message.clone());
+			self.multicast_inner(protocol, message_hash, topic, || message.clone());
 			Some((topic, message))
 		} else {
 			trace!(target:"gossip", "Ignored statement from unregistered peer {}", who);
 			None
 		}
 	}
 
+
+	/// Handle an incoming ConsensusStatusMessage. Forwards it to the consensus engine handler.
+	pub fn on_status(
+		&mut self,
+		protocol: &mut Context<B>,
+		who: NodeIndex,
+		message: ConsensusStatusMessage,
+	) {
+		if let Some(_) = self.peers.get_mut(&who) {
+			let engine_id = message.engine_id;
+			//validate the message
+			if let Some(validator) = self.validators.get(&engine_id) {
+				validator.on_status(who, &message.data)
```

Comment:
shouldn't there be some context here that lets the validator say that now certain messages should be send to that peer?

Reply:
use-case: peer is previously at GRANDPA round 5 and we are at round 10.
peer sends us a status message indicating that he is now at round 10. So we want to send round 10 messages that he was previously ignoring.

Reply:
Added `StatusResult`